config/v1/types_oauth.go

```diff
@@ -107,17 +120,6 @@ type OAuthIdentityProvider struct {
 	// +optional
 	MappingMethod MappingMethodType `json:"mappingMethod"`
 
-	// grantMethod: allow, deny, prompt
-	// This method will be used only if the specific OAuth client doesn't provide a strategy
-	// of their own. Valid grant handling methods are:
-	//  - auto:   always approves grant requests, useful for trusted clients
-	//  - prompt: prompts the end user for approval of grant requests, useful for third-party clients
-	//  - deny:   always denies grant requests, useful for black-listed clients
-	// Defaults to "prompt" if not set.
-	// +optional
-	GrantMethod GrantHandlerType `json:"grantMethod"`
```

Comment:
if @ericavonb signs off on the extra migration pain (two releases I think) and you keep this as an option to your binary so that if a customer uses it they aren't completely stuck, then removing the option here can work.

Reply:
> if @ericavonb signs off on the extra migration pain (two releases I think) and you keep this as an option to your binary so that if a customer uses it they aren't completely stuck, then removing the option here can work.

Oh, and you'll need to expose a metric that indicates if you had to set it during migration so we know if we have to add it back.

Reply:
this is a migration across a major version, so I do not think this concern holds.

Reply:
i would not block the pr on this issue, we can add it back if required over the beta.